packages/wrangler/src/deployment-bundle/esbuild-plugins/als-external.ts

```diff
@@ -0,0 +1,13 @@
+import type { Plugin } from "esbuild";
+
+/**
+ * An esbuild plugin that will mark `node:async_hooks` imports as external.
+ */
+export const asyncLocalStoragePlugin: Plugin = {
+	name: "Mark async local storage imports as external plugin",
+	setup(pluginBuild) {
+		pluginBuild.onResolve({ filter: /^node:async_hooks/ }, () => {
```

Comment:
```suggestion
		pluginBuild.onResolve({ filter: /^node:async_hooks$/ }, () => {
```

Reply:
I did this for the potential future case of there being a sub entry point to `async_hooks`. Since we would also want this plugin to kick in for imports like:

```ts
import {...} from "node:async_hooks/blah";
```

Do you feel that we should not support this @RamIdeas ?

Reply:
I guess there is the fear that a new module called `node:async_hooks_2` would also match this...

I could tighten this up a bit...?

Reply:
I was thinking the latter but I guess the former is equally likely. Since we're only marking as external, I think we can leave it as-is 👍

Reply:
Doh! I updated it now 😆